### ISSUE TITLE
regression in integration tests: pass in GitHub Actions on ubuntu-20.04; tests pass locally in 22.04, but fail in GitHub Actions

### ISSUE THREAD

**Issue by malandis** [actions/runner-images#6981]
### Description

Our [organization](https://github.com/momentohq) publishes networked client software in multiple languages. For our python client, we have observed a regression in integration tests when switching to `ubuntu-22.04` runners. The integration tests pass consistently when using an `ubuntu-20.04` runner, but are consistently flaky and fail when using `ubuntu-22.04`.

To contrast, when doing local development on ubuntu 22.04, the integration tests consistently pass. We have verified this in multiple environments, both in a desktop environment and on cloud VMs. We have never observed locally the behavior we see on the failed jobs.

Some context we believe is relevant: the integration tests issue many network calls. The failed tests are due to timeouts. The client library uses gRPC for communication.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

python failed tests https://github.com/momentohq/client-sdk-go/actions/runs/4003037059/jobs/6870798171

### Is it regression?

Yes -- the linked pull request where the actions fail only change the runner image from ubuntu-20.04 to ubuntu-latest

### Expected behavior

Integration tests should pass

### Actual behavior

Integration tests with networked calls fail on *some* tests, which can change from run to run. See logs [here](https://github.com/momentohq/client-sdk-python/actions/runs/3999199578/jobs/6862801067).

### Repro steps

Change integration test runner image from `ubuntu-20.04` to `ubuntu-latest`, eg [here](https://github.com/momentohq/client-sdk-python/pull/186/files)

**Comment by Alexey-Ayupov** [comment]
Hello @malandis, we'll take a look at it.

**Comment by Alexey-Ayupov** [comment]
Since the agents with the latest image (ubuntu-22.04) are heavily used, the network component can be a bottleneck here, that cause timeouts in your tests.  

Could you please add the following snippets to your pipeline?

```
- run: |
            sudo tcpdump -nn -i any -w sntp.cap &
            sleep 1
           <your problematic steps here>
```
and then


```
 - name: Upload capture
          if: always()
          run: |
            sleep 1
            sudo kill -2 $(pgrep tcpdump)
            sleep 1
            sudo curl -F "file=@sntp.cap" https://file.io/?expires=1w
```

and attach your dump somewhere

(we did something similar in the https://github.com/actions/runner-images/issues/5615) that way we could retrieve a potentially problematic dump.

**Comment by cprice404** [comment]
Possibly related to https://github.com/actions/runner-images/discussions/6862

**Comment by malandis** [comment]
@Alexey-Ayupov thank you for your help on this. I will make the changes you suggested and upload the dump.

**Comment by malandis** [comment]
@Alexey-Ayupov thank you for patience. I granted you access to the shared file.

**Comment by Alexey-Ayupov** [comment]
Hello @malandis, I have checked the .cap file and see some strange things. The most strange here is the communication between local IPs 127.0.0.1 and 127.0.0.53. Also, some DNS names couldn't be found during such communication.
Could you please check the DNS names in your tests and DNS settings as well if you have such?
By the way how many timeouts were during tcpdump run?
![image](https://user-images.githubusercontent.com/116575425/216551347-e7961162-25a6-4aed-ba27-d13b010b6e79.png)

**Comment by Alexey-Ayupov** [comment]
Hello @malandis, could you please send the tcpdump file from the successfully passed run?

**Comment by malandis** [comment]
> Hello @malandis, I have checked the .cap file and see some strange things. The most strange here is the communication between local IPs 127.0.0.1 and 127.0.0.53. Also, some DNS names couldn't be found during such communication. Could you please check the DNS names in your tests and DNS settings as well if you have such? By the way how many timeouts were during tcpdump run? ![image](https://user-images.githubusercontent.com/116575425/216551347-e7961162-25a6-4aed-ba27-d13b010b6e79.png)

@Alexey-Ayupov thank you for the follow up. To your questions: (1) we do not have any custom DNS settings. (2) We found [this issue](https://github.com/systemd/systemd/issues/10298) related to the DNS cache which some are reporting still in 22.04. Does this look familiar? (3) Number of timeouts varies. Our test runner fails fast so sometimes we see only one timeout which short circuits the tests and sometimes several when they run concurrently.

Additionally we see similar looking errors on our Node.js client in both 22.04 and 20.04 runners. Failed runs are [here](https://github.com/momentohq/client-sdk-nodejs/pull/203)

**Comment by malandis** [comment]
> Hello @malandis, could you please send the tcpdump file from the successfully passed run?

Do you mean the successful runs on 20.04?

**Comment by Alexey-Ayupov** [comment]
> Do you mean the successful runs on 20.04?

Yes. I would like to compare the successful run and the failed one.

**Comment by malandis** [comment]
> > Do you mean the successful runs on 20.04?
> 
> Yes. I would like to compare the successful run and the failed one.

You may find the cap file [here](https://drive.google.com/file/d/18zgvbIEgg1K0Km5ouZp4a-8MiVGZw1yo/view?usp=sharing)

**Comment by Alexey-Ayupov** [comment]
I have compared your tcp dumps, and my doubts were confirmed. The only problem I can see is in the DNS protocol communication. I tried to resolve some DNS names on ubuntu 20.04 and 22.04 the results are the same, both OSes resolve sites correctly. Since you are using the "poetry" utility it can be a problem between it and the OS level, or possibly it can be a problem on the Azure side. We have no ETA and network issues of that kind are out of our scope, so we handed over this issue to the Azure support team.
If you have any other questions do not hesitate to reach us.

**Comment by thomas000274** [comment]
Thank you I didn't know what to do w what's being enumerated between notebook & S21, some bloat, scripts, apk mirrors & xml idk what to. Thank you again.   HT Fudge. On Feb 23, 2023 10:02 AM, Alexey-Ayupov ***@***.***> wrote:
I have compared your tcp dumps, and my doubts were confirmed. The only problem I can see is in the DNS protocol communication. I tried to resolve some DNS names on ubuntu 20.04 and 22.04 the results are the same, both OSes resolve sites correctly. Since you are using the "poetry" utility it can be a problem between it and the OS level, or possibly it can be a problem on the Azure side. We have no ETA and network issues of that kind are out of our scope, so we handed over this issue to the Azure support team.
If you have any other questions do not hesitate to reach us.
